device/vlk/vlk_pass.cpp

```diff
@@ -238,22 +238,28 @@ namespace RayGene3D
 
 
       {
+        auto layers = uint32_t(1);
+
         std::vector<VkImageView> rt_attachment_views(rt_views.size());
         for (size_t i = 0; i < rt_views.size(); ++i)
         {
           const auto& rt_view = rt_views[i];
           rt_attachment_views[i] = (reinterpret_cast<VLKView*>(rt_view.get()))->GetView();
-          extent_x = std::max(extent_x, rt_view->GetResource().GetExtentX());
-          extent_y = std::max(extent_y, rt_view->GetResource().GetExtentY());
+          extent_x = std::max(extent_x, rt_view->GetResource().GetExtentX() >> rt_view->GetMipmapOffset());
+          extent_y = std::max(extent_y, rt_view->GetResource().GetExtentY() >> rt_view->GetMipmapOffset());
+          const auto layer_count = rt_view->GetLayerCount();
+          layers = std::max(layer_count == -1 ? 1 : layer_count, layers);
         }
 
         std::vector<VkImageView> ds_attachment_views(ds_views.size());
         for (size_t i = 0; i < ds_views.size(); ++i)
         {
           const auto& ds_view = ds_views[i];
           ds_attachment_views[i] = (reinterpret_cast<VLKView*>(ds_view.get()))->GetView();
-          extent_x = std::max(extent_x, ds_view->GetResource().GetExtentX());
-          extent_y = std::max(extent_y, ds_view->GetResource().GetExtentY());
+          extent_x = std::max(extent_x, ds_view->GetResource().GetExtentX() >> ds_view->GetMipmapOffset());
```

Comment:
Let us add two methods to set up both extent_x/extent_y
We can further validate rt/ds size against these values